include/util/typedefs.hpp

```diff
@@ -58,6 +58,9 @@ using NodeID = std::uint32_t;
 using EdgeID = std::uint32_t;
 using NameID = std::uint32_t;
 using EdgeWeight = std::int32_t;
+using EdgeDuration = std::int32_t;
+using SegmentWeight = std::int32_t;
```

Comment:
here and below can be used unsigned value. 

also unpacking should extend sign if type is signed. it is not used atm but better to add a static assert that the base type of the packed vector is unsigned